Sources/IO/Geometry/STLWriter/index.js

```diff
@@ -0,0 +1,216 @@
+    const strips = polyData.getStrips() ? polyData.getStrips().getData() : null;
+    let file = '';
+
+    file += 'solid ascii\n';
+
+    const n = [];
+    let v1 = [];
+    let v2 = [];
+    let v3 = [];
+
+    // Strips
+
+    if (strips && strips.length > 0) {
+      throw new Error('Unsupported strips');
+    }
+
+    // Polys
+    for (let i = 0; i < polys.length; ) {
+      const pointNumber = polys[i++];
+
+      if (pointNumber) {
+        v1 = [
+          points[polys[i] * 3],
+          points[polys[i] * 3 + 1],
+          points[polys[i++] * 3 + 2],
+        ];
+        v2 = [
+          points[polys[i] * 3],
+          points[polys[i] * 3 + 1],
+          points[polys[i++] * 3 + 2],
+        ];
+        v3 = [
+          points[polys[i] * 3],
+          points[polys[i] * 3 + 1],
+          points[polys[i++] * 3 + 2],
+        ];
+        if (transform) {
+          vec3.transformMat4(v1, v1, transform);
+          vec3.transformMat4(v2, v2, transform);
+          vec3.transformMat4(v3, v3, transform);
+        }
+
+        vtkTriangle.computeNormal(v1, v2, v3, n);
+
+        file += ` facet normal ${n[0].toPrecision(6)} ${n[1].toPrecision(
+          6
+        )} ${n[2].toPrecision(6)}\n`;
+        file += '  outer loop\n';
+        file += `   vertex ${v1[0].toPrecision(6)} ${v1[1].toPrecision(
+          6
+        )} ${v1[2].toPrecision(6)}\n`;
+        file += `   vertex ${v2[0].toPrecision(6)} ${v2[1].toPrecision(
+          6
+        )} ${v2[2].toPrecision(6)}\n`;
+        file += `   vertex ${v3[0].toPrecision(6)} ${v3[1].toPrecision(
+          6
+        )} ${v3[2].toPrecision(6)}\n`;
+        file += '  endloop\n';
+        file += ' endfacet\n';
+      } else {
+        throw new Error('Unsupported polygon');
+      }
+    }
+
+    file += 'endsolid\n';
+
+    return file;
+  }
+
+  publicAPI.requestData = (inData, outData) => {
+    const input = inData[0];
+    if (!input || input.getClassName() !== 'vtkPolyData') {
+      vtkErrorMacro('Invalid or missing input');
+      return;
+    }
+    if (model.format === FormatTypes.ASCII) {
+      outData[0] = writeASCII(input, model.transform);
+    } else if (model.format === FormatTypes.BINARY) {
+      outData[0] = writeBinary(input, model.transform);
+    } else {
+      vtkErrorMacro('Invalid format type');
+    }
+  };
+}
+
+// ----------------------------------------------------------------------------
+// Object factory
+// ----------------------------------------------------------------------------
+
+const DEFAULT_VALUES = {
+  format: FormatTypes.BINARY,
+  transform: null,
+};
+
+// ----------------------------------------------------------------------------
+
+export function extend(publicAPI, model, initialValues = {}) {
+  Object.assign(model, DEFAULT_VALUES, initialValues);
+
+  // Make this a VTK object
+  macro.obj(publicAPI, model);
+
+  // Also make it an algorithm with one input and one output
+  macro.algo(publicAPI, model, 1, 1);
```

Comment:
it is an algorithm with 0 output .

See ZipMultiDataSetWriter and XMLWriter

Reply:
I thought that was elegant to provide the data as output. Especially for the web context.

We could expose some helper that provide the output via a different method name. But I like that approach.

Reply:
Gotcha, so the other writers should probably have the same change (does not have to be done in this PR)

Reply:
Yes, I would think so to have some consistency. 

Then we may want to have helper methods to get the output like:

```
getOutputAsBlob();

// Only for binary format
getOutputAsArrayBuffer() {
   setEncodingAsBinary();
   return getOutputData();
}

// Only for/when ascii format
getOuptutAsString() { 
   setEncodingAsString();
   return getOutputData();
}
```

But like you said, that could be done via a refactoring pass on all the writer we have. 
No need to do it in that PR.